lib/utils.js

```diff
@@ -45,4 +45,28 @@ utils.removeHopToHopHeaders = function removeHopToHopHeaders(rh, removeContentEn
     });
 };
 
+/**
+ * From a list of uri Regex and values, constructs a regex to check if the
+ * request URI is in the white-list.
+ */
+var CACHE = new Map();
+utils.constructRegex = function(variants) {
+    if (CACHE.has(variants)) {
+        return CACHE.get(variants);
+    }
+    var regex = (variants || []).map(function(regexString) {
```

Comment:
Perhaps worth adding a .trim() to get rid of leading / trailing ws on regexString.